src/encoded/audit/annotation.py

```diff
@@ -19,3 +20,20 @@ def audit_annotation_organism(value, system):
     if 'organism' not in value:
         detail = ('Annotation {} lacks organism information.'.format(audit_link(path_to_text(value['@id']), value['@id'])))
         raise AuditFailure('missing organism', detail, level='INTERNAL_ACTION')
+
+
+@audit_checker('Annotation')
+def audit_annotation_derived_from_revoked(value, system):
+    '''
+    Annotations with files derived from a revoked file should be flagged with an audit.
+    '''
+    request = system.get('request')
+    for file in value.get('files'):
+        if 'derived_from' in file:
+            for f in file['derived_from']:
+                parent = request.embed(f + '@@object')
+                if parent['status'] == 'revoked':
```

Comment:
I thikn you should use 
`parent = request.embed(f + '@@object?skip_calculated=true')`
instead.